### PR TITLE
Timer tdd

### PR DESCRIPTION
**Pull request by peterdu98**
Using TDD for refactoring Timer.
Using TDD for adding the new Counter class.
Fixing bug that the timer won't stop when the game is over. (it can stop now)

### CODE REVIEW

**Review by dannamachi — CHANGES_REQUESTED** [review]
Add your test comments

File: ducksteroid/src/Timer.cs [hunk 1]
```diff
@@ -6,53 +6,72 @@ namespace MyGame.src
...
             _lastcalled = TimeInSec;
         }
 
         //Properties
         public int LastCalledSec { get => _lastcalled; set => _lastcalled = value; }
-        public int TimeInSec { get { return _min * 60 + _sec; } }
-        public int Min { get => _min; }
-        public int Sec { get => _sec; }
+        public int TimeInSec { get { return _min.Value * 60 + _sec.Value; } }
+        private Counter Hr { get => _hr; }
```
Comment (dannamachi) [inline ducksteroid/src/Timer.cs:27]: why do you need these private properties if you can just access them normally within the class

**Review by dannamachi — CHANGES_REQUESTED** [review]
comment

File: ducksteroid/src/Timer.cs (hunk 1 above)
Comment (peterdu98) [reply ducksteroid/src/Timer.cs:27]: It's more clear when using the filed.

Comment (dannamachi) [reply ducksteroid/src/Timer.cs:27]: its not rlly necessary but if u say so

**Review by dannamachi — APPROVED** [review]

**Pull request merged**